syntax = "proto2";
package MWNN;

message MWNNTensorProto {
  optional string name = 1;                      //Name of the Tensor
  optional int32 type = 2;                       //MWNN Datatype of the Tensor
  repeated int32 dims = 3;                       //Shape of the Tensor
  repeated float float_data = 4 [packed = true]; //Values of the Tensor
}

message MWNNAttributeProto {
  optional string name = 1;        //Name of the Attribute
  optional int32 type = 2;         //MWNN Datatype of the Attribute
  repeated int32 int_data = 3;     //Int values of the Attribute
  repeated float float_data = 4;   //Float values of the Attribute
  repeated string string_data = 5; //String values of the Attribute
}

message MWNNNodeProto {
  optional string name = 1;                  // Name of the Node
  optional string op_type = 2;               // OpType of the node
  repeated string ip_name = 3;               // Node Inputs name
  repeated string op_name = 4;               // Node Outputs name
  repeated MWNNAttributeProto attribute = 5; // Node Attributes
}

message MWNNGraphProto {
  optional string name = 1;                 // Name of the graph
  repeated string ip_name = 2;              // Graph Inputs name
  repeated string op_name = 3;              // Graph Outputs name
  repeated MWNNNodeProto node = 4;          // Graph Nodes
  repeated MWNNTensorProto initializer = 5; // Graph Constant Initializers
  repeated MWNNTensorProto input = 6;       // Graph Input Tensor
  repeated MWNNTensorProto output = 7;      // Graph Output Tensor
}
